cupyx/scipy/sparse/_coo.py

```diff
@@ -414,7 +415,18 @@ def sum_duplicates(self):
             data = cupy.zeros(size, dtype=self.data.dtype)
             row = cupy.empty(size, dtype='i')
             col = cupy.empty(size, dtype='i')
-            if self.data.dtype.kind == 'f':
+            if self.data.dtype.kind == 'b':
+                cupy.ElementwiseKernel(
+                    'T src_data, int32 src_row, int32 src_col, int32 index',
+                    'raw T data, raw int32 row, raw int32 col',
+                    '''
+                    if (src_data) data[index] = true;
```

Comment:
```suggestion
                    if (src_data) data[index] = true;
```
Guess this is also related to #7724 :)